### PR TITLE
Add recipe for the coastal-dune-model

### PR DESCRIPTION
**Pull request by mcflugen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mcflugen** [comment]
@conda-forge/help-c-cpp This is ready for review now. Thanks!

I had to set *PKG_CONFIG_PATH* in *bld.bat* to include *%LIBRARY_LIB%\pkgconfig*. Is that normal? It seems a little hacky. I would have thought the *pkg-config* from *conda* would already know about that folder (maybe the system, rather than the *conda*, *pkg-config* is being used?).

**Comment by mcflugen** [comment]
@conda-forge-admin, please ping team

File: recipes/cdm/build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+mkdir -p _build && cd _build
+
+cmake ../src \
+  -DCMAKE_INSTALL_PREFIX=$PREFIX \
+  -DCMAKE_BUILD_TYPE=Release
+
```
Comment (ocefpaf) [inline recipes/cdm/build.sh:6]: Is there a `make check/test` we can run here?

Comment (mcflugen) [reply recipes/cdm/build.sh:6]: There isn't yet, unfortunately. I'm working with the main author to put some better tests together, though.

(A grad student is waiting for a binary version of this package, which is why I've put this together before adding tests. I hope that's ok.)

**Comment by mcflugen** [comment]
Thanks @ocefpaf!

**Comment by ocefpaf** [comment]
> (A grad student is waiting for a binary version of this package, which is why I've put this together before adding tests. I hope that's ok.)

A grad student life is already too hard. If this package will make their life a bit easier we have to merge it!

**Comment by isuruf** [comment]
> A grad student life is already too hard. If this package will make their life a bit easier we have to merge it!

:smile:

**Pull request merged**